src/test/regress/specs/isolation_failure_testing.spec

```diff
@@ -0,0 +1,37 @@
+# see that we can cause a session to block at a point of our choosing
+
+session "s1"
+
+setup { BEGIN; SET deadlock_timeout TO '100ms'; }
+step "s1-lock-1" { SELECT pg_advisory_xact_lock(1); }
+step "s1-lock-2" { SELECT pg_advisory_xact_lock(2); }
+step "s1-commit" { COMMIT; }
+
+session "s2"
+
+setup { BEGIN; SET deadlock_timeout TO '100ms'; SET application_name TO 'gdb-process' }
+step "s2-lock-1" { SELECT pg_advisory_xact_lock(1); }
+step "s2-lock-2" { SELECT pg_advisory_xact_lock(2); }
+step "s2-commit" { COMMIT; }
+
+step "s2-lock-1-gdb"
+{
+  SELECT 1;
+}
+
+session "gdb"
+
+step "gdb-break"
+{
+  WITH pid AS (
+    SELECT pid FROM pg_stat_activity WHERE application_name = 'gdb-process'
+  ) SELECT citus.gdb_attach(pid.pid) FROM pid;
+
+  SELECT citus.run_command('!break NeedsDistributedPlanning');
+  SELECT citus.run_command('continue');
```

Comment:
The API seems a bit messy at the moment. Would it make sense to do something like:
```
SELECT citus.gdb_init();
SELECT citus.gdb_break('NeedsDistributedPlanning');
SELECT citus.gdb_continue();
```

Reply:
Yeah, 100%. This API isn't very user friendly. I haven't made it nicer yet for a couple reasons:
- I'm not sure what the right API is yet, I want to write some more tests before deciding what the right set of concepts is.
- These functions are a little hard to add and iterate on. I haven't figured out how to share code between the functions so I need to copy and paste the code for connecting to a socket and modify it in all of them whenever it changes. For now just having a `run_command` is easier to develop with.